tests/lib/client.py

```diff
@@ -108,6 +141,27 @@ def provision_migration_instance(
             query_string={"accepts_incomplete": accepts_incomplete},
         )
 
+    def update_instance(
+        self,
+        instance_model: (
```

Comment:
same note about type hints here - probably `type[ServiceInstance]` 
it might be worth making a type alias like:
```
InstanceModel = type[ServiceInstance]
<...>
def update_instance(
    self,
    instance_model: InstanceModel,
<...>

```